src/main/java/internet/store/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,48 @@
+package internet.store.service.impl;
+
+import internet.store.dao.ShopCartDao;
+import internet.store.lib.Inject;
+import internet.store.lib.Service;
+import internet.store.model.Product;
+import internet.store.model.ShoppingCart;
+import internet.store.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShopCartDao shopCartDao;
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        return shopCartDao.create(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(product);
+        return shopCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public boolean deleteProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().remove(product);
+        shopCartDao.update(shoppingCart);
+        return !shoppingCart.getProducts().removeIf(p -> p.equals(product));
```

Comment:
I don't understand what this does

Reply:
Return "true" if there are no such product anymore in the Cart (i.e. operation in line 29 was successful and attempt to remove this product second time is failed)